### PR TITLE
chore(Accordion): Convert to RFC

### PR DESCRIPTION
**Pull request by assuncaocharles**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

# BREAKING CHANGES

This PR converts `Accordion` component to be functional. Not passing any prop to styles functions.

Related to #12237

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/fluentui/pull/12876)

### CODE REVIEW

**Comment by assuncaocharles** [comment]
If I understood correctly, screener is complaining that before active elements didn't had outline and now they have ( has it suppose to be ). So I think it should be fine, @layershifter ?

File: packages/fluentui/react-northstar/src/components/Accordion/Accordion.tsx [hunk 1]
```diff
@@ -84,114 +95,108 @@ export interface AccordionProps extends UIComponentProps, ChildrenComponentProps
...
+      design,
+      styles,
+      variables,
+    }),
+    rtl: context.rtl,
+  });
 
-    this.focusHandler = new ContainerFocusHandler(this.getNavigationItemsSize, this.handleNavigationFocus, true);
-  }
+  const [focusedIndex, setfocusedIndex] = React.useState(0);
```
Comment (silviuaavram) [inline packages/fluentui/react-northstar/src/components/Accordion/Accordion.tsx:173]: not sure if we want this to be initialised to 0. won't the first item always be focused at first render?

File: packages/fluentui/react-northstar/src/components/Accordion/Accordion.tsx [hunk 2]
```diff
@@ -84,114 +95,108 @@ export interface AccordionProps extends UIComponentProps, ChildrenComponentProps
...
-    const { exclusive } = this.props;
+  const focusHandler: ContainerFocusHandler = new ContainerFocusHandler(
+    getNavigationItemsSize,
+    handleNavigationFocus,
+    true,
+  );
 
-    if (!this.isIndexActionable(index)) {
+  let itemRefs = [];
+  const defaultAccordionTitleId = _.uniqueId('accordion-title-');
```
Comment (silviuaavram) [inline packages/fluentui/react-northstar/src/components/Accordion/Accordion.tsx:195]: can you check if these get re-generated at every render? if so, maybe use a ref or memo.

File: packages/fluentui/react-northstar/src/components/Accordion/Accordion.tsx [hunk 3]
```diff
@@ -235,46 +237,40 @@ class Accordion extends AutoControlledComponent<WithAsProp<AccordionProps>, Acco
...
-    this.itemRefs = [];
-    this.focusHandler.syncFocusedIndex(focusedIndex);
 
+    itemRefs = [];
+    focusHandler.syncFocusedIndex(focusedIndex);
     _.each(panels, (panel, index) => {
       const { content, title } = panel;
-      const active = this.isIndexActive(index);
-      const canBeCollapsed = this.isIndexActionable(index);
+      const active = isIndexActive(+index);
```
Comment (silviuaavram) [inline packages/fluentui/react-northstar/src/components/Accordion/Accordion.tsx:255]: what's the thing with that `+` ?

File: packages/fluentui/react-northstar/test/specs/components/Accordion/Accordion-test.tsx [hunk 4]
```diff
@@ -107,7 +132,7 @@ describe('Accordion', () => {
       const wrapper = mountWithProvider(<Accordion panels={panels} />);
       const accordion = wrapper.find(Accordion);
       getTitleButtonAtIndex(wrapper, 1).simulate('click');
-      expect(accordion.state('focusedIndex')).toEqual(1);
+      expect(getAccordionTitleAtIndex(accordion, 1)).toEqual(document.activeElement);
```
Comment (silviuaavram) [inline packages/fluentui/react-northstar/test/specs/components/Accordion/Accordion-test.tsx:135]: check this out https://github.com/testing-library/jest-dom#tohavefocus :D

File: packages/fluentui/react-northstar/src/components/Accordion/Accordion.tsx [hunk 5]
```diff
@@ -235,46 +237,40 @@ class Accordion extends AutoControlledComponent<WithAsProp<AccordionProps>, Acco
...
+      const contentId = content['id'] || `${defaultAccordionContentId}${index}`;
+      itemRefs[index] = contentRef;
 
       children.push(
         createShorthand(AccordionTitle, title, {
           defaultProps: () => ({
             className: accordionSlotClassNames.title,
             active,
-            index,
+            index: +index,
```
[the remark this answers is not in the document]
Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/components/Accordion/Accordion.tsx:267]: It's just converting to int `"0"` -> `0 `

File: packages/fluentui/react-northstar/src/components/Accordion/Accordion.tsx (hunk 1 above)
Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/components/Accordion/Accordion.tsx:173]: You are right. I am changing to use the value of `activeIndex` in the initialization

File: packages/fluentui/react-northstar/src/components/Accordion/Accordion.tsx (hunk 2 above)
Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/components/Accordion/Accordion.tsx:195]: Changed to `useMemo`

File: packages/fluentui/react-northstar/src/components/Accordion/Accordion.tsx (hunk 3 above)
Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/components/Accordion/Accordion.tsx:255]: Parse to int

File: packages/fluentui/react-northstar/test/specs/components/Accordion/Accordion-test.tsx (hunk 4 above)
Comment (assuncaocharles) [reply packages/fluentui/react-northstar/test/specs/components/Accordion/Accordion-test.tsx:135]: Nice, changed all

**Pull request merged**